load-json.js

```diff
@@ -2,26 +2,45 @@ var loadedJSON = null;
 
 inputField = document.querySelector("#JSONFile");
 inputField.onchange = function () {
+    echartslayer.chart.showLoading();   // Show Loading Animation
+
     if ('files' in inputField && inputField.files.length == 1) {
         var file = inputField.files[0];
         var reader = new FileReader();
-        reader.onload = function(e)
-        {
+        reader.onload = function (e) {
             var contents = e.target.result;
-            try
-            {
+            try {
                 loadedJSON = JSON.parse(contents.toString());
             }
-            catch (e)
-            {
+            catch (e) {
                 alert("Unable to parse JSON");
             }
             console.log(loadedJSON);
+
+            // create loc data array
+            var locationData = [];
+            for(var i=0; i<loadedJSON.locations.length; i++){

```

Comment:
This should fail with an alert message if there is no "locations" array or if the "locations" array is not of the correct format.

Reply:
Fixed.